src/server/models/listItem/listItem.ts

```diff
@@ -388,12 +439,80 @@ export async function update(
   }
 }
 
+/**
+ * Updates the isAnnualReview flag for list items and adds a ListItemEvent record.
+ * @param listItems
+ * @param status
+ * @param eventName
+ * @param auditEvent
+ */
+export async function updateIsAnnualReview(
+  list: List,
+  listItems: ListItemWithHistory[],
+  listItemEvent: ListItemEvent,
+  eventName: AuditListItemEventName,
+  auditEvent: AuditEvent
+): Promise<Result<ListItemWithHistory[]>> {
+  const updatedListItems: ListItemWithHistory[] = [];
+
+  if (!listItems) {
+    const message = `List ids must be provided to update list items`;
+    logger.error(message);
+    return { error: new Error(message) };
+  }
+  for (const listItem of listItems) {
+    const updateListItemPrismaStatement: Prisma.ListItemUpdateArgs = {
+      where: {
+        id: listItem.id,
+      },
+      data: {
+        isAnnualReview: listItem.status !== Status.UNPUBLISHED,
```

Comment:
this is a very specific function. I'm not sure why we're passing in the parameters that we are, like 
```
  listItemEvent: ListItemEvent,
  eventName: AuditListItemEventName,
  auditEvent: AuditEvent
```

it suggests the event can be flexible, but we are just setting annual review here. so there's only one event that should happen ListItemEvent.ANNUAL_REVIEW_STARTED.

this should suffice: 
```
history: {
          create: EVENTS[ListItemEvent.ANNUAL_REVIEW_STARTED](),
         },

```

similar for the audit update `recordListItemUpdate`. I don't think any other event type should be passed in here. Let's make it just a plain object, and we can get rid of the clutter in the parameters.

--

I'm not sure why the array of listItems is passed in as with history -- do we need it?

-- 

I'm also not sure why this is an iteration - it's several writes to the database. I think I would probably prefer the whole thing in a transaction. If one update fails, they all fail, we fix the issue and retry the next day. 


this should do most of what you need apart from error handling and you can forego the need for iterating. 
```.ts
  const annualReviewRef = list.jsonData.currentAnnualReview?.reference;
  const listItemIds = listItems.map((listItem) => listItem.id);

  // although this is probably best as an updateMany as well
  const query = prisma.listItem.updateMany({
    where: {
      id: {
        in: listItemIds,
      },
      isPublished: true,
      history: {
        some: {
          type: Status.PUBLISHED,
          time: {
            lte: subDays(new Date(), 30), 
          },
        },
      },
    },
    data: {
      isAnnualReview: true,
      history: {
        create: EVENTS[ListItemEvent.ANNUAL_REVIEW_STARTED](),
      },
    },
  });

  await prisma.$transaction([query, ...recordEvents])

```